operators/vision/encode_image.hpp

```diff
@@ -8,23 +8,148 @@
+      }
+    }
+
+    return {};
+  }
+
+  OrtStatusPtr OnModelAttach(const OrtApi& api, const OrtKernelInfo& info) {
+    std::unordered_map<std::string, std::variant<std::string>> attrs = {
+        {"color_space", "bgr"},
+        {"file_extension", "png"}
+    };
+
+    std::string format;
+    auto status = OrtW::API::GetOpAttributeString(api, info, "format", format);
+    if (status != nullptr) {
+      return status;
+    }
+    if (!format.empty()){
+      if (format != "jpg" && format != "png") {
+        return OrtW::CreateStatus("[EncodeImage] 'format' attribute value must be 'jpg' or 'png'.", ORT_RUNTIME_EXCEPTION);
+      } else {
+        attrs["file_extension"] = std::string(".") + format;
+      }
     }
 
-    extension_ = std::string(".") + format;
+    std::string clr;
+    status = OrtW::API::GetOpAttributeString(api, info, "color_space", clr);
+    if (status != nullptr) {
+      return status;
+    }
+    if (!clr.empty()) {
+      if (clr != "bgr" && clr != "rgb") {
+        return OrtW::CreateStatus("[EncodeImage] 'color_space' attribute value must be 'bgr' or 'rgb'.", ORT_RUNTIME_EXCEPTION);
+      } else {
+        attrs["color_space"] = clr;
+      }
+    }
+
+    return Init(attrs);
   }
 
-  void Compute(const ortc::Tensor<uint8_t>& input_bgr,
-               ortc::Tensor<uint8_t>& output) const;
+  OrtxStatus Compute(const ortc::Tensor<uint8_t>& input, ortc::Tensor<uint8_t>& output) const{
+    const auto& dimensions = input.Shape();
+    if (dimensions.size() != 3 || dimensions[2] != 3) {
+      return {kOrtxErrorInvalidArgument, "[EncodeImage] requires rank 3 rgb input in channels last format."};
+    }
+
+    int32_t height = static_cast<int32_t>(dimensions[0]);  // H
+    int32_t width = static_cast<int32_t>(dimensions[1]);   // W
+    const int32_t color_space = 3;
+    const uint8_t* input_data_ptr = input.Data();
+    uint8_t* outbuffer = nullptr;
+    size_t outsize = 0;
+
+    std::unique_ptr<uint8_t[]> conversion_buf;
+    auto rgb_data = input_data_ptr;
+    bool conversion_needed = false;
+    auto fx_supported_bgr = 
+      extension_ == ".png"? &internal::EncodeImage::pngSupportsBgr : &internal::EncodeImage::JpgSupportsBgr;
+
+    bool bgr_source = is_bgr_;  
+    if ((this->*fx_supported_bgr)() != is_bgr_) {
+      conversion_needed = true;
+      bgr_source = !is_bgr_;
+    }
+
+    if (conversion_needed) {
+      conversion_buf = std::make_unique<uint8_t[]>(height * width * color_space);
+      auto cvt_data = conversion_buf.get();
+      for (int32_t y = 0; y < height; ++y) {
+        for (int32_t x = 0; x < width; ++x) {
+          cvt_data[(y * width + x) * color_space + 0] = input_data_ptr[(y * width + x) * color_space + 2];
+          cvt_data[(y * width + x) * color_space + 1] = input_data_ptr[(y * width + x) * color_space + 1];
+          cvt_data[(y * width + x) * color_space + 2] = input_data_ptr[(y * width + x) * color_space + 0];
+        }
+      }
+      rgb_data = cvt_data;
```

Comment:
Is it correct `rgb_data` is not really used? I think if `fx_supported_bgr` if false and `is_bgr_` is true, the `rgb_data` should be used instead of `input_data_ptr`, right?

Reply:
you are right, there is a missing change on calling EncodeJPG/PNG function.